### PR TITLE
bump roapi to 0.9.0 and columnq-cli to 0.4.0

### LINKED ISSUE #240: Unify native support for S3, GCS and Azure using object_store
### Background
roapi supports S3 before Datafusion adopts crate `object_store`. And it uses `rusoto_s3` to preload S3 files to memory.

https://github.com/roapi/roapi/blob/68ee99a92491379c1d90845d9a807b43e1bde79a/columnq/src/io/s3.rs#L54

### Problem
1. S3 with `use_memory_table: false` fails
2. Added native GCS/Azure support may duplicate work with S3, when different client SDK required

### Expect
1. support S3 with both memory table and datafusion table
2. Easily support native GCS and Azure

### Proposal
Each action item can be a separate PR
- [x] fix S3 with `use_memory_table: false` and use ObjectStoreProvider and global SessionContext: https://github.com/roapi/roapi/pull/241
- [x] replace `rusoto_s3` (in `use_memory_table: true` path) with S3 client from `object_store` (`s3://`)
    - [x] https://github.com/roapi/roapi/pull/244
    - [x] https://github.com/roapi/roapi/pull/245
    - [x] https://github.com/roapi/roapi/pull/247
    - [x] ~release `v0.8.2`~
- [x] extend ObjectStoreProvider to support GCS with `object_store` (`gs://`)
    - [x] https://github.com/roapi/roapi/pull/248
    - [x] https://github.com/roapi/docs/pull/16
- [x] extend ObjectStoreProvider to support Azure with `object_store` (`az://`, `adl://`, `abfs://`, `abfss://`, `azure://`)
    - [x] https://github.com/roapi/roapi/pull/250
    - [x] https://github.com/roapi/docs/pull/17
- [ ] release `v0.9.0`: https://github.com/roapi/roapi/pull/251

### Related
There are several action items proposed. If all done, this issue can close
- https://github.com/roapi/roapi/issues/227
- https://github.com/roapi/roapi/issues/186
- https://github.com/roapi/roapi/issues/121

### PR DESCRIPTION
**Pull request by jychen7**
close #240

### CODE REVIEW

File: roapi/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "roapi"
-version = "0.8.1"
+version = "0.9.0"
```
Comment (jychen7) [inline roapi/Cargo.toml:3]: do we want to bump `columnq-cli` to `0.4.0` as well?

https://github.com/roapi/roapi/blob/2ec39ddba8fff359be1d8273bf43e942df421007/columnq-cli/Cargo.toml#L1-L3

Comment (houqp) [reply roapi/Cargo.toml:3]: yeah, let's do that

**Review by houqp — APPROVED** [review]

**Review by houqp — APPROVED** [review]

File: roapi/Cargo.toml (hunk 1 above)
Comment (jychen7) [reply roapi/Cargo.toml:3]: sounds good, done

**Comment by jychen7** [comment]
> Merging is blocked
The base branch restricts merging to authorized users.

will left to you, @houqp :D

**Comment by houqp** [comment]
that's weird, I will look into fixing that.

**Comment by houqp** [comment]
updated the repo setting, let's see if it works for the next PR.

**Comment by jychen7** [comment]
> Let's bump the version and aim for a new release this weekend?

hi @houqp , I just notice PR merge will not trigger a release, until we create a `tag`. Would you like to create a tag or you would like me to push a tag? And want to do it in weekday or wait until weekend?

**Comment by houqp** [comment]
@jychen7 feel free to push a tag to trigger the release. I was planning to wait for the fix for https://github.com/roapi/roapi/issues/174, but you don't need to wait for it if you want to push a release now. We can always release with a subsequent patch version bump.

**Comment by jychen7** [comment]
created tag `roapi-v0.9.0` and `columnq-cli-v0.4.0` for main (bc5754a4ed6d5cfef3605e83402cf6070cd6f4bb)

**Comment by houqp** [comment]
@jychen7 thanks! Just FYI, for future releases, you can click the `generate release  note` button on the release page to populate a detailed change log ;)

**Comment by jychen7** [comment]
👍

**Pull request merged**